drivers/boston_project_scorecard/app/models/boston_project_scorecard/report.rb

```diff
@@ -0,0 +1,219 @@
+
+    belongs_to :project, class_name: 'GrdaWarehouse::Hud::Project', optional: true
+    belongs_to :project_group, class_name: 'GrdaWarehouse::ProjectGroup', optional: true
+    belongs_to :user, class_name: 'User', optional: true
+    belongs_to :apr, class_name: 'HudReports::ReportInstance', optional: true
+
+    has_many :project_contacts, through: :project, source: :contacts
+    has_many :organization_contacts, through: :project
+    has_many :project_group_project_contacts, through: :project_group, source: :contacts
+    has_many :project_group_organization_contacts, through: :project_group, source: :organization_contacts
+
+    scope :started_between, ->(start_date:, end_date:) do
+      where(started_at: (start_date..end_date))
+    end
+
+    def completed?
+      status == 'completed'
+    end
+
+    def pending?
+      status == 'pending'
+    end
+
+    def locked?(field, _user)
+      # TODO: Implement field access rules for users
+      case status
+      when 'pending'
+        case field
+        when :period_start_date, :period_end_date, :project_type, :secondary_reviewer_id
+          # Allow editing the header while being pre-filled
+          false
+        else
+          true
+        end
+
+      when 'pre-filled'
+        # Entire form is editable after pre-filling
+        false
+
+      when 'ready'
+        case field
+        when :period_start_date, :period_end_date, :project_type, :secondary_reviewer_id
+          # Lock header when ready
+          true
+        else
+          false
+        end
+
+      else
+        true
+      end
+    end
+
+    def admin?(user)
+      # The user that created the report is the admin
+      user_id == user.id
+    end
+
+    def field_input_options(field, user)
+      if locked?(field, user)
+        { readonly: true }
+      else
+        {}
+      end
+    end
+
+    private def score(value, ten_range, five_range = nil)
+      return nil if value.blank?
+
+      if ten_range.include?(value)
+        10
+      elsif five_range.present? && five_range.include?(value)
+        5
+      else
+        0
+      end
+    end
+
+    private def percentage(value)
+      return 0 if value.nan?
+      return 0 if value.infinite?
+
+      (value * 100).to_i
+    end
+
+    def controlled_parameters
+      @controlled_parameters ||= [
+        :project_type,
+        :period_start_date,
+        :period_end_date,
+        :secondary_reviewer,
+        :initial_goals_pass,
+        :initial_goals_notes,
+        :timeliness_pass,
+        :timeliness_notes,
+        :independent_living_pass,
+        :independent_living_notes,
+        :management_oversight_pass,
+        :management_oversight_notes,
+        :prioritization_pass,
+        :prioritization_notes,
+      ].freeze
+    end
+
+    def project_name
+      return project.name if project.present?
+
+      project_group.name
+    end
+
+    def title
+      'MA-500 Boston Continuum of Care FY21 Renewal Project Scoring Tool'
```

Comment:
This should probably be:
```suggestion
      _('MA-500 Boston Continuum of Care FY21 Renewal Project Scoring Tool')
```